{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This file serves to test the python i would further use in the snakefile without running the whole pipeline\n",
    "## Test 1 : read results and create a table for datapane"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from glob import glob\n",
    "import altair as alt\n",
    "import csv\n",
    "dirname = input('What\\'s the folder name honey')\n",
    "path = '/mnt/R60-11/Bacterio/metagenomique_NextSeq/' + dirname + '/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>13D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>kitomeD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>11C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>13C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>kitomeC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TPOS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>TENV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GODOC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>FLOCH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HEZER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>LHOSTIS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Description\n",
       "Sample_ID            \n",
       "1                 11D\n",
       "2                 12D\n",
       "3                 13D\n",
       "4             kitomeD\n",
       "5                 11C\n",
       "6                 12C\n",
       "7                 13C\n",
       "8             kitomeC\n",
       "9                TPOS\n",
       "10               TENV\n",
       "11              GODOC\n",
       "12              FLOCH\n",
       "13              HEZER\n",
       "14            LHOSTIS"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(path + 'SampleSheet.csv', 'r') as csvfile:\n",
    "    lines = csvfile.readlines()\n",
    "sections = [lines.index(line) for line in lines if \"[\" in line]\n",
    "SampleSheet = lines[sections[3]+2:]\n",
    "header = lines[sections[3]+1].split(sep=\",\")\n",
    "SampleSheet = pd.DataFrame([line.split(\",\") for line in SampleSheet], columns = header)[['Sample_ID','Description']].set_index('Sample_ID')\n",
    "SampleSheet\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = '/mnt/R60-11/Bacterio/metagenomique_NextSeq/' + dirname + '/results/temp/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>Description</th>\n",
       "      <th>raw</th>\n",
       "      <th>quality</th>\n",
       "      <th>nonHuman</th>\n",
       "      <th>percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11D</td>\n",
       "      <td>1821360</td>\n",
       "      <td>1148155</td>\n",
       "      <td>6492</td>\n",
       "      <td>0.356437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>12D</td>\n",
       "      <td>2148734</td>\n",
       "      <td>1369402</td>\n",
       "      <td>4471</td>\n",
       "      <td>0.208076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>13D</td>\n",
       "      <td>2203017</td>\n",
       "      <td>1477499</td>\n",
       "      <td>16872</td>\n",
       "      <td>0.765859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>kitomeD</td>\n",
       "      <td>263174</td>\n",
       "      <td>99032</td>\n",
       "      <td>28690</td>\n",
       "      <td>10.901533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>11C</td>\n",
       "      <td>1792333</td>\n",
       "      <td>1234955</td>\n",
       "      <td>2049</td>\n",
       "      <td>0.114320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>12C</td>\n",
       "      <td>6942381</td>\n",
       "      <td>5324485</td>\n",
       "      <td>41338</td>\n",
       "      <td>0.595444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>13C</td>\n",
       "      <td>5362957</td>\n",
       "      <td>4059166</td>\n",
       "      <td>41276</td>\n",
       "      <td>0.769650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>kitomeC</td>\n",
       "      <td>7662826</td>\n",
       "      <td>5455151</td>\n",
       "      <td>607278</td>\n",
       "      <td>7.924987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>TPOS</td>\n",
       "      <td>1455065</td>\n",
       "      <td>1220981</td>\n",
       "      <td>1220015</td>\n",
       "      <td>83.846082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>TENV</td>\n",
       "      <td>446</td>\n",
       "      <td>167</td>\n",
       "      <td>117</td>\n",
       "      <td>26.233184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>GODOC</td>\n",
       "      <td>2778381</td>\n",
       "      <td>2260516</td>\n",
       "      <td>314</td>\n",
       "      <td>0.011302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>FLOCH</td>\n",
       "      <td>256311</td>\n",
       "      <td>144751</td>\n",
       "      <td>12522</td>\n",
       "      <td>4.885471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>HEZER</td>\n",
       "      <td>794132</td>\n",
       "      <td>465776</td>\n",
       "      <td>218</td>\n",
       "      <td>0.027451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>LHOSTIS</td>\n",
       "      <td>6456316</td>\n",
       "      <td>5103219</td>\n",
       "      <td>11225</td>\n",
       "      <td>0.173861</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Sample Description      raw  quality  nonHuman    percent\n",
       "1        1         11D  1821360  1148155      6492   0.356437\n",
       "2        2         12D  2148734  1369402      4471   0.208076\n",
       "3        3         13D  2203017  1477499     16872   0.765859\n",
       "4        4     kitomeD   263174    99032     28690  10.901533\n",
       "5        5         11C  1792333  1234955      2049   0.114320\n",
       "6        6         12C  6942381  5324485     41338   0.595444\n",
       "7        7         13C  5362957  4059166     41276   0.769650\n",
       "8        8     kitomeC  7662826  5455151    607278   7.924987\n",
       "9        9        TPOS  1455065  1220981   1220015  83.846082\n",
       "10      10        TENV      446      167       117  26.233184\n",
       "11      11       GODOC  2778381  2260516       314   0.011302\n",
       "12      12       FLOCH   256311   144751     12522   4.885471\n",
       "13      13       HEZER   794132   465776       218   0.027451\n",
       "14      14     LHOSTIS  6456316  5103219     11225   0.173861"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "D = pd.read_table(path + 'Nreads_raw.txt' ,sep=\",\",names=['raw'],index_col=0)\n",
    "D = D.join(pd.read_table(path + 'Nreads_qc.txt' ,sep=\",\",names=['quality'],index_col=0))\n",
    "D = D.join(pd.read_table(path + 'Nreads_nothumain_1.txt' ,sep=\",\",names=['human1'],index_col=0))\n",
    "D = D.join(pd.read_table(path + 'Nreads_nothumain_2.txt' ,sep=\",\",names=['human2'],index_col=0))\n",
    "D['nonHuman'] = D[\"human1\"] - D[\"human2\"]\n",
    "D['percent'] = D['nonHuman'] / D['raw'] * 100\n",
    "D = D.drop(labels=\"Undetermined\", axis=0)\n",
    "D =D.join(SampleSheet)\n",
    "D = D[['Description','raw','quality','nonHuman','percent']].sort_index()\n",
    "D.index = D.index.astype(int)\n",
    "D = D[['Description','raw','quality','nonHuman','percent']].sort_index()\n",
    "D['Sample'] = D.index\n",
    "D[['Sample','Description','raw','quality','nonHuman','percent']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>steps</th>\n",
       "      <th>reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>raw</td>\n",
       "      <td>3431382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>raw</td>\n",
       "      <td>972512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>raw</td>\n",
       "      <td>54706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>raw</td>\n",
       "      <td>33124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>raw</td>\n",
       "      <td>1290077</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Sample steps    reads\n",
       "0       1   raw  3431382\n",
       "1       2   raw   972512\n",
       "2       3   raw    54706\n",
       "3       4   raw    33124\n",
       "4       5   raw  1290077"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "D1 = pd.melt(D[['Sample','raw','nonHuman']], id_vars=['Sample'], value_vars=['raw','nonHuman'], var_name='steps', value_name='reads')\n",
    "D1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<div id=\"altair-viz-05aa2f76f47645ed9af672623fee17c4\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-05aa2f76f47645ed9af672623fee17c4\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-05aa2f76f47645ed9af672623fee17c4\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm//vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm//vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm//vega-lite@4.17.0?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm//vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"4.17.0\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 400, \"continuousHeight\": 300}, \"axis\": {\"labelFontSize\": 15, \"titleFontSize\": 20}}, \"data\": {\"name\": \"data-39422c6b068da8be6772988308ad6a11\"}, \"mark\": \"line\", \"encoding\": {\"column\": {\"field\": \"steps\", \"type\": \"nominal\"}, \"tooltip\": [{\"field\": \"reads\", \"type\": \"quantitative\"}, {\"field\": \"Sample\", \"type\": \"nominal\"}], \"x\": {\"field\": \"Sample\", \"type\": \"nominal\"}, \"y\": {\"field\": \"reads\", \"type\": \"quantitative\"}}, \"height\": 300, \"width\": 1200, \"$schema\": \"https://vega.github.io/schema/vega-lite/v4.17.0.json\", \"datasets\": {\"data-39422c6b068da8be6772988308ad6a11\": [{\"Sample\": 1, \"steps\": \"raw\", \"reads\": 3431382}, {\"Sample\": 2, \"steps\": \"raw\", \"reads\": 972512}, {\"Sample\": 3, \"steps\": \"raw\", \"reads\": 54706}, {\"Sample\": 4, \"steps\": \"raw\", \"reads\": 33124}, {\"Sample\": 5, \"steps\": \"raw\", \"reads\": 1290077}, {\"Sample\": 6, \"steps\": \"raw\", \"reads\": 195124}, {\"Sample\": 7, \"steps\": \"raw\", \"reads\": 522}, {\"Sample\": 8, \"steps\": \"raw\", \"reads\": 1316425}, {\"Sample\": 9, \"steps\": \"raw\", \"reads\": 15257157}, {\"Sample\": 10, \"steps\": \"raw\", \"reads\": 360601}, {\"Sample\": 11, \"steps\": \"raw\", \"reads\": 6074690}, {\"Sample\": 12, \"steps\": \"raw\", \"reads\": 3223067}, {\"Sample\": 13, \"steps\": \"raw\", \"reads\": 2190436}, {\"Sample\": 1, \"steps\": \"nonHuman\", \"reads\": 263}, {\"Sample\": 2, \"steps\": \"nonHuman\", \"reads\": 201960}, {\"Sample\": 3, \"steps\": \"nonHuman\", \"reads\": 2434}, {\"Sample\": 4, \"steps\": \"nonHuman\", \"reads\": 12335}, {\"Sample\": 5, \"steps\": \"nonHuman\", \"reads\": 128}, {\"Sample\": 6, \"steps\": \"nonHuman\", \"reads\": 66929}, {\"Sample\": 7, \"steps\": \"nonHuman\", \"reads\": 143}, {\"Sample\": 8, \"steps\": \"nonHuman\", \"reads\": 1031045}, {\"Sample\": 9, \"steps\": \"nonHuman\", \"reads\": 50249}, {\"Sample\": 10, \"steps\": \"nonHuman\", \"reads\": 86361}, {\"Sample\": 11, \"steps\": \"nonHuman\", \"reads\": 619}, {\"Sample\": 12, \"steps\": \"nonHuman\", \"reads\": 179203}, {\"Sample\": 13, \"steps\": \"nonHuman\", \"reads\": 338}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Plot t\n",
    "alt.Chart(D1).mark_line().encode(\n",
    "            x='Sample:N',\n",
    "            y='reads:Q',\n",
    "            column='steps:N',\n",
    "            tooltip = [alt.Tooltip('reads:Q'),alt.Tooltip('Sample:N')]\n",
    "        ).properties(width=1200,\n",
    "                    height=300\n",
    "        ).configure_axis(labelFontSize=15,\n",
    "                         titleFontSize=20\n",
    "        )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Test 2 : read kraken results and create a table for datapane"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys \n",
    "import os\n",
    "import argparse\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = '/mnt/R60-11/Bacterio/metagenomique_NextSeq/' + dirname + '/results/03_not_human/report/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "def input_dir(dir):\n",
    "\tfiles=os.listdir(dir)\n",
    "\treport=[]\n",
    "\tfor f in files:\n",
    "\t\tfipath=os.path.join(dir,f)\n",
    "\t\treport.append(fipath)\t\n",
    "\treturn (report)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "filelist=input_dir(path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "alldf = []\n",
    "for file in filelist:\n",
    "    if 'Undetermined' in file:\n",
    "        continue\n",
    "    out = pd.read_csv(file, sep=\"\\t\", names=['percent','reads_root','reads_assigned','rank', 'taxon_id','name'])\n",
    "    out = out[out[\"rank\"] == \"S\"]\n",
    "    out[\"name\"] = out[\"name\"].str.strip()\n",
    "    out[\"Sample\"] = file.split(\"/\")[-1].split(\".\")[0]\n",
    "    alldf.append(out)\n",
    "    superdf = pd.concat(alldf)\n",
    "    table = superdf[[\"name\",\"reads_root\", \"Sample\"]]\n",
    "    table = table.pivot_table(index=\"name\", columns=\"Sample\", values=\"reads_root\")\n",
    "    table = table.drop(labels=\"Homo sapiens\", axis=0)\n",
    "    table = table.fillna(0)\n",
    "    table = table[table.columns.astype(int).sort_values().astype(str)]\n",
    "    table.loc[:,'Total'] = table.sum(axis=1)\n",
    "    table = table[table[\"Total\"] > 100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Sample</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Actinomyces oris</th>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>629.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>640.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Alcanivorax sp. N3-2A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4545.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4564.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anoxybacillus flavithermus</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>326.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Atopobium parvulum</th>\n",
       "      <td>26.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>159.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bacillus cereus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>583.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Streptococcus sp. oral taxon 431</th>\n",
       "      <td>116.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>250.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1470.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1890.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Streptococcus viridans</th>\n",
       "      <td>23.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>88.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>221.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thermoanaerobacterium thermosaccharolyticum</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>219.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>223.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Veillonella dispar</th>\n",
       "      <td>377.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>989.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>277.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1849.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Veillonella parvula</th>\n",
       "      <td>309.0</td>\n",
       "      <td>198.0</td>\n",
       "      <td>841.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1582.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>81 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Sample                                           1      2      3    4    5  \\\n",
       "name                                                                         \n",
       "Actinomyces oris                               0.0    2.0    1.0  1.0  0.0   \n",
       "Alcanivorax sp. N3-2A                          0.0    0.0    0.0  0.0  2.0   \n",
       "Anoxybacillus flavithermus                     0.0    0.0    0.0  0.0  0.0   \n",
       "Atopobium parvulum                            26.0   24.0   69.0  2.0  0.0   \n",
       "Bacillus cereus                                1.0    0.0    0.0  2.0  5.0   \n",
       "...                                            ...    ...    ...  ...  ...   \n",
       "Streptococcus sp. oral taxon 431             116.0   51.0  250.0  1.0  0.0   \n",
       "Streptococcus viridans                        23.0   14.0   88.0  0.0  0.0   \n",
       "Thermoanaerobacterium thermosaccharolyticum    0.0    0.0    2.0  0.0  0.0   \n",
       "Veillonella dispar                           377.0  198.0  989.0  8.0  0.0   \n",
       "Veillonella parvula                          309.0  198.0  841.0  5.0  0.0   \n",
       "\n",
       "Sample                                           6     7       8      9   10  \\\n",
       "name                                                                           \n",
       "Actinomyces oris                               3.0   4.0   629.0    0.0  0.0   \n",
       "Alcanivorax sp. N3-2A                         10.0   6.0  4545.0    1.0  0.0   \n",
       "Anoxybacillus flavithermus                     0.0   0.0   326.0    0.0  0.0   \n",
       "Atopobium parvulum                             0.0   0.0    38.0    0.0  0.0   \n",
       "Bacillus cereus                              122.0  62.0    25.0  364.0  0.0   \n",
       "...                                            ...   ...     ...    ...  ...   \n",
       "Streptococcus sp. oral taxon 431               0.0   0.0  1470.0    0.0  0.0   \n",
       "Streptococcus viridans                         0.0   0.0   221.0    0.0  0.0   \n",
       "Thermoanaerobacterium thermosaccharolyticum    1.0   1.0   219.0    0.0  0.0   \n",
       "Veillonella dispar                             0.0   0.0   277.0    0.0  0.0   \n",
       "Veillonella parvula                            1.0   0.0   227.0    0.0  0.0   \n",
       "\n",
       "Sample                                        11   12   13   14   Total  \n",
       "name                                                                     \n",
       "Actinomyces oris                             0.0  0.0  0.0  0.0   640.0  \n",
       "Alcanivorax sp. N3-2A                        0.0  0.0  0.0  0.0  4564.0  \n",
       "Anoxybacillus flavithermus                   0.0  0.0  0.0  0.0   326.0  \n",
       "Atopobium parvulum                           0.0  0.0  0.0  0.0   159.0  \n",
       "Bacillus cereus                              0.0  1.0  0.0  1.0   583.0  \n",
       "...                                          ...  ...  ...  ...     ...  \n",
       "Streptococcus sp. oral taxon 431             0.0  2.0  0.0  0.0  1890.0  \n",
       "Streptococcus viridans                       0.0  0.0  0.0  0.0   346.0  \n",
       "Thermoanaerobacterium thermosaccharolyticum  0.0  0.0  0.0  0.0   223.0  \n",
       "Veillonella dispar                           0.0  0.0  0.0  0.0  1849.0  \n",
       "Veillonella parvula                          0.0  1.0  0.0  0.0  1582.0  \n",
       "\n",
       "[81 rows x 15 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "31f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6"
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
